awx/main/scheduler/kubernetes.py

```diff
@@ -0,0 +1,150 @@
+import time
+import yaml
+import tempfile
+from base64 import b64encode
+
+from django.conf import settings
+from kubernetes import client, config
+from django.utils.functional import cached_property
+
+from awx.main.utils.common import parse_yaml_or_json
+
+
+class PodManager(object):
+
+    def __init__(self, task=None):
+        self.task = task
+
+    def deploy(self):
+        if not self.credential.kubernetes:
+            raise RuntimeError('Pod deployment cannot occur without a Kubernetes credential')
+
+
+        self.kube_api.create_namespaced_pod(body=self.pod_definition,
+                                            namespace=self.namespace,
+                                            _request_timeout=settings.AWX_CONTAINER_GROUP_DEFAULT_LAUNCH_TIMEOUT)
+
+        # We don't do any fancy timeout logic here because it is handled
+        # at a higher level in the job spawning process. See
+        # settings.AWX_ISOLATED_LAUNCH_TIMEOUT and settings.AWX_ISOLATED_CONNECTION_TIMEOUT
```

Comment:
Is this statement actually true?  This code is called from the `def run` in `tasks.py`, and it seems to me that if the pod status got stuck in pending forever, this code would just happily hang out in this loop until the end of time (which would probably be bad).

Reply:
Maybe we should put some timer code into the `while True:` loop just to be safe e.g.,

```python
start = time.time()
while True:
if time.time() - start > AWX_CONTAINER_GROUP_DEFAULT_LAUNCH_TIMEOUT:
    raise RuntimeError(...)
```

Reply:
Hrm, I wonder if I wrote this when it was called inside of the task manager.

@ryanpetrello since I'm going to move this into a task, is there a more generic way of handling a timeout when the worker works on stuff?